### PR TITLE
pkp/pkp-lib#429 For lib/pkp submodule, respect announcement publication date

### PR DESCRIPTION
**Pull request by asmecher**

### CODE REVIEW

File: classes/announcement/PKPAnnouncementDAO.inc.php [hunk 1]
```diff
@@ -339,6 +339,7 @@ function &getAnnouncementsNotExpiredByAssocId($assocType, $assocId, $rangeInfo =
 			WHERE assoc_type = ?
 				AND assoc_id = ?
 				AND (date_expire IS NULL OR date_expire > CURRENT_DATE)
+				AND (date_posted < CURRENT_DATE)
```
Comment (beghelli) [inline classes/announcement/PKPAnnouncementDAO.inc.php]: is not null would make more sense right? Can we set the date posted to anything higher than  the current date?

File: classes/announcement/PKPAnnouncementDAO.inc.php [hunk 2]
```diff
@@ -370,16 +372,18 @@ function &getNumAnnouncementsNotExpiredByAssocId($assocType, $assocId, $numAnnou
 	}
 
 	/**
-	 * Retrieve most recent announcement by Assoc ID.
+	 * Retrieve most recent published announcement by Assoc ID.
 	 * @param $assocType int
 	 * @return Announcement
 	 */
-	function &getMostRecentAnnouncementByAssocId($assocType, $assocId) {
+	function &getMostRecentPublishedAnnouncementByAssocId($assocType, $assocId) {
```
Comment (beghelli) [inline classes/announcement/PKPAnnouncementDAO.inc.php]: I think we don't need the reference anymore...

File: classes/announcement/PKPAnnouncementDAO.inc.php (hunk 1 above)
Comment (asmecher) [inline classes/announcement/PKPAnnouncementDAO.inc.php]: Bruno, are you talking about the date_expired check or the date_posted one?

File: classes/announcement/PKPAnnouncementDAO.inc.php (hunk 2 above)
Comment (asmecher) [inline classes/announcement/PKPAnnouncementDAO.inc.php]: This is for OJS 2.4.x, and I'd rather not change reference semantics until 3.0. While technically I suspect we no longer have good PHP4 compatibility, there's no sense in disturbing stable code.

Comment (beghelli) [inline classes/announcement/PKPAnnouncementDAO.inc.php]: oups, I was thinking about master, sorry

File: classes/announcement/PKPAnnouncementDAO.inc.php (hunk 1 above)
Comment (beghelli) [inline classes/announcement/PKPAnnouncementDAO.inc.php]: date_posted one

Comment (asmecher) [inline classes/announcement/PKPAnnouncementDAO.inc.php]: The bug reporter's goal was to permit setting a future date and having the announcement not get published until then.

**Pull request merged**